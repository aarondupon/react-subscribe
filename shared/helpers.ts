export function decapitalize(s:string):string{
    if (typeof s !== 'string') return ''
    return s.charAt(0).toLocaleLowerCase() + s.slice(1)
}