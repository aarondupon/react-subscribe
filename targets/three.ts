import React,{ReactType} from 'react';
import * as three from 'three';
import {decapitalize} from '../shared/helpers'
import createSubscriberComponent from '../subscribe/createSubscriberComponent'



export const apply = (target,camelCase = true)=>{
    let extendedSubscribed = createSubscriberComponent;
     Object.keys(target).map((key)=>{
       const Component:string = camelCase ? decapitalize(key) : key;
      extendedSubscribed[Component] = createSubscriberComponent(Component as ReactType);   
     })
     return extendedSubscribed;
};

let extendedSubscribed = apply(three);

export {
    extendedSubscribed as subscribe,
    extendedSubscribed as s,
 }