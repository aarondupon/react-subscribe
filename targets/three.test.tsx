import React,{ReactType,useState} from 'react';
import * as three from 'three';
import {decapitalize} from '../shared/helpers'
import {apply} from './three';
import {subscribe} from './three';
import renderer from 'react-test-renderer';
import {renderer as renderThree, Canvas} from 'react-three-fiber';
import Enzyme,{mount,render,shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ArDisplay from '../../view/ArDisplay'

Enzyme.configure({ adapter: new Adapter() })

let extendedSubscribed = apply(three);

const Button = (props)=>(<a onClick={props.onClick} href="f">aarera</a>)
const RenderSigninForm = () => {
    const [isDisabled, setDisabled] = useState(false)
    return(<form>
      <Button  
        onClick={()=>setDisabled(true)}
        disabled={isDisabled}
      >Login</Button>
    </form>)
};

 describe('extend subscrib JSX', () => {
    it('test apply (three', () => {
        // extendedSubscribed = apply({Geometry:three.Geometry})
        expect(subscribe).toBeTruthy() 
        expect(three.Geometry).toBeTruthy() 
        expect(subscribe.geometry).toBeTruthy()

        
    });

    it('renders correctly_', () => {
        const tree = renderer
          .create(React.createElement('subscribe.geometry'))
          .toJSON();
        expect(tree).toMatchSnapshot();
        console.log(tree)
      });

        it('renders <subscribe.geometry />', () => {
          const tree = renderer.create(
            <subscribe.geometry />
          ).toJSON();
          expect(tree.type).toBe('geometry');
        });
      
        it('renders <subscribe.geometry /> with color', () => {
          const tree = renderer.create(
            <subscribe.geometry  fuck={'red'}/>
          ).toJSON();
          console.log(tree)
          expect(tree.props.fuck).toBe('red');
        });
      });
      
      
  })
 
