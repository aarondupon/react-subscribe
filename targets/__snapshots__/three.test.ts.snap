// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extend subscrib JSX renders correctly 1`] = `<subscribe.geometry />`;

exports[`extend subscrib JSX renders correctly_ 1`] = `<subscribe.geometry />`;

exports[`extend subscrib JSX renders correctly2 1`] = `<subscribe.ffgeometry />`;
